### PR TITLE
fix(tracing): prioritize `RUST_LOG` and verbosity CLI arg

### LINKED ISSUE #6326: `RUST_LOG` doesn't affect the stdout log
### Describe the bug

`RUST_LOG` should apply the directive to all configured logging outputs. Currently, it applies at least to the file logging, but not for the stdout.

### Steps to reproduce

-

### Node logs

```text
➜  reth (main) RUST_LOG=trace cargo run --bin reth init --chain mainnet                                                                                                                                                  ✭
    Finished dev [unoptimized + debuginfo] target(s) in 1.46s
     Running `target/debug/reth init --chain mainnet`
2024-02-01T15:51:47.247820Z  INFO reth::cli: reth init starting
2024-02-01T15:51:47.247917Z  INFO reth::cli: Opening database path="/Users/shekhirin/Library/Application Support/reth/mainnet/db"
2024-02-01T15:51:47.253023Z  INFO reth::cli: Database opened
2024-02-01T15:51:47.253052Z  INFO reth::cli: Writing genesis block
2024-02-01T15:51:47.269411Z  INFO reth::cli: Genesis block written hash=0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3

➜  reth (main) tail ~/Library/Caches/reth/logs/mainnet/reth.log                                                                                                                                                          ✭
2024-02-01T15:46:15.671726Z DEBUG reth::init: Genesis already written, skipping.
2024-02-01T15:46:15.687107Z  INFO reth::cli: Genesis block written hash=0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3
2024-02-01T15:51:47.245550Z  INFO reth::cli: bello info
2024-02-01T15:51:47.245592Z DEBUG reth::cli: bello debug
2024-02-01T15:51:47.247858Z  INFO reth::cli: reth init starting
2024-02-01T15:51:47.247960Z  INFO reth::cli: Opening database path="/Users/shekhirin/Library/Application Support/reth/mainnet/db"
2024-02-01T15:51:47.253038Z  INFO reth::cli: Database opened
2024-02-01T15:51:47.253056Z  INFO reth::cli: Writing genesis block
2024-02-01T15:51:47.253708Z DEBUG reth::init: Genesis already written, skipping.
2024-02-01T15:51:47.269511Z  INFO reth::cli: Genesis block written hash=0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3
```

### Platform(s)

Mac (Apple Silicon)

### What version/commit are you on?

0.1.0-alpha.16 9f82af2

### What database version are you on?

1

### What type of node are you running?

Archive (default)

### What prune config do you use, if any?

_No response_

### If you've built Reth from source, provide the full command you used

_No response_

### Code of Conduct

- [X] I agree to follow the Code of Conduct

### PR DESCRIPTION
**Pull request by shekhirin**
Fixes https://github.com/paradigmxyz/reth/issues/6326

```console
➜  reth (alexey/fix-tracing-priorities) ./target/debug/reth init                                                                                                                                                                                                                                                              ✭
2024-02-05T17:21:22.212039Z  INFO reth::cli: reth init starting
2024-02-05T17:21:22.212139Z  INFO reth::cli: Opening database path="/Users/shekhirin/Library/Application Support/reth/mainnet/db"
2024-02-05T17:21:22.227596Z  INFO reth::cli: Database opened
2024-02-05T17:21:22.227637Z  INFO reth::cli: Writing genesis block
2024-02-05T17:21:22.242580Z  INFO reth::cli: Genesis block written hash=0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3

➜  reth (alexey/fix-tracing-priorities) RUST_LOG=trace,mio=debug ./target/debug/reth init                                                                                                                                                                                                                                     ✭
2024-02-05T17:21:33.462368Z  INFO reth::cli: reth init starting
2024-02-05T17:21:33.462472Z  INFO reth::cli: Opening database path="/Users/shekhirin/Library/Application Support/reth/mainnet/db"
2024-02-05T17:21:33.477912Z  INFO reth::cli: Database opened
2024-02-05T17:21:33.477950Z  INFO reth::cli: Writing genesis block
2024-02-05T17:21:33.478123Z DEBUG reth::init: Genesis already written, skipping.
2024-02-05T17:21:33.492974Z  INFO reth::cli: Genesis block written hash=0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3

➜  reth (alexey/fix-tracing-priorities) ./target/debug/reth init -vvvvv                                                                                                                                                                                                                                                       ✭
2024-02-05T17:21:39.010038Z TRACE mio::poll: registering event source with poller: token=Token(1), interests=READABLE    
2024-02-05T17:21:39.010741Z  INFO reth::cli: reth init starting
2024-02-05T17:21:39.010827Z  INFO reth::cli: Opening database path="/Users/shekhirin/Library/Application Support/reth/mainnet/db"
2024-02-05T17:21:39.024427Z  INFO reth::cli: Database opened
2024-02-05T17:21:39.024457Z  INFO reth::cli: Writing genesis block
2024-02-05T17:21:39.024558Z DEBUG reth::init: Genesis already written, skipping.
2024-02-05T17:21:39.039156Z  INFO reth::cli: Genesis block written hash=0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3

➜  reth (alexey/fix-tracing-priorities) ./target/debug/reth init --log.stdout.filter trace,mio=debug                                                                                                                                                                                                                          ✭
2024-02-05T17:21:42.424145Z  INFO reth::cli: reth init starting
2024-02-05T17:21:42.424235Z  INFO reth::cli: Opening database path="/Users/shekhirin/Library/Application Support/reth/mainnet/db"
2024-02-05T17:21:42.438824Z  INFO reth::cli: Database opened
2024-02-05T17:21:42.438861Z  INFO reth::cli: Writing genesis block
2024-02-05T17:21:42.438963Z DEBUG reth::init: Genesis already written, skipping.
2024-02-05T17:21:42.453986Z  INFO reth::cli: Genesis block written hash=0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3

➜  reth (alexey/fix-tracing-priorities) RUST_LOG=trace ./target/debug/reth init --log.stdout.filter trace,mio=debug                                                                                                                                                                                                           ✭
2024-02-05T17:21:46.420171Z  INFO reth::cli: reth init starting
2024-02-05T17:21:46.420277Z  INFO reth::cli: Opening database path="/Users/shekhirin/Library/Application Support/reth/mainnet/db"
2024-02-05T17:21:46.434365Z  INFO reth::cli: Database opened
2024-02-05T17:21:46.434409Z  INFO reth::cli: Writing genesis block
2024-02-05T17:21:46.434515Z DEBUG reth::init: Genesis already written, skipping.
2024-02-05T17:21:46.449367Z  INFO reth::cli: Genesis block written hash=0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3

➜  reth (alexey/fix-tracing-priorities) RUST_LOG=trace ./target/debug/reth init -vvvvv                                                                                                                                                                                                                                        ✭
2024-02-05T17:21:54.362538Z TRACE mio::poll: registering event source with poller: token=Token(1), interests=READABLE    
2024-02-05T17:21:54.363205Z  INFO reth::cli: reth init starting
2024-02-05T17:21:54.363296Z  INFO reth::cli: Opening database path="/Users/shekhirin/Library/Application Support/reth/mainnet/db"
2024-02-05T17:21:54.376522Z  INFO reth::cli: Database opened
2024-02-05T17:21:54.376559Z  INFO reth::cli: Writing genesis block
2024-02-05T17:21:54.376660Z DEBUG reth::init: Genesis already written, skipping.
2024-02-05T17:21:54.391456Z  INFO reth::cli: Genesis block written hash=0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3

```

### CODE REVIEW

File: crates/tracing/src/lib.rs [hunk 1]
```diff
@@ -19,8 +19,8 @@
 //!  fn main() -> eyre::Result<()> {
 //!      let tracer = RethTracer::new().with_stdout(LayerInfo::new(
 //!          LogFormat::Json,
+//!          LevelFilter::INFO.to_string(),
```
Comment (emhane) [inline crates/tracing/src/lib.rs:22]: agreed that info should be default

Comment (emhane) [reply crates/tracing/src/lib.rs:22]: or hang on, what does it mean that info and debug swapped places as `filters` and `directive`?

File: crates/tracing/src/lib.rs [hunk 2]
```diff
@@ -138,16 +136,16 @@ impl LayerInfo {
     ///      - `LogFormat::Json` for JSON formatting.
     ///      - `LogFormat::LogFmt` for logfmt (key=value) formatting.
     ///      - `LogFormat::Terminal` for human-readable, terminal-friendly formatting.
+    ///  * `defaut_directive` - Directive for filtering log messages.
```
Comment (emhane) [inline crates/tracing/src/lib.rs:139]:
```suggestion
    ///  * `default_directive` - Directive for filtering log messages.
```

**Review by emhane — APPROVED** [review]
lgtm

**Review by rkrasiuk — APPROVED** [review]

**Review by mattsse — APPROVED** [review]

**Pull request merged**